docs/voiceover-guidelines.md

```diff
@@ -65,7 +183,220 @@ contentView.accessibilityElements = [subtotalTitleLabel,
+
+```swift
+let snackBarView = createSnackBarView()
+presentSnackBar(snackBarView)
+
+UIAccessibility.post(notification: .layoutChanged, argument: snackBarView)
+```
+
+This notifies VoiceOver that a new view, `snackBarView`, has appeared and it should move the _focus_ to it. VoiceOver will then read its derived `accessibilityLabel`.
+
+Once the element disappears, you should send another notification but with a `nil` argument. This makes VoiceOver immediately move the focus to a different element on the screen.
+
+```swift
+UIAccessibility.post(notification: .layoutChanged, argument: nil)
+```
+
+### <a name="prefer-disabling"></a>Prefer Disabling Instead of Hiding Elements
+
+A common UI pattern is hiding elements, such as buttons, until users are able to use them. 
+
+<img src="images/voiceover-guidelines/hidden-button.gif" width="320">
+
+This is not ideal for vision accessibility. VoiceOver users, especially first-time users, may try to understand how to interact with your UI by navigating through all the elements on the screen. If an important button, such as the **Insert button** example above, is initially hidden, they may never discover it. This can make comprehension difficult.
+
+Consider making the element visible, but disabled, instead. 
+
+```swift
+func viewDidLoad() {
+    // Allow VoiceOver users to discover the Insert button. But don't let them use it yet.
+    insertButton.isEnabled = false
+}
+
+func onImageSelected() {
+    // Enable the Insert button when an image was selected.
+    insertButton.isEnabled = true
+}
+```
+
+If a `UIControl`'s `isEnabled` property is set to `false`, UIKit would, by default, automatically add the [`.notEnabled` accessibility trait](https://developer.apple.com/documentation/uikit/uiaccessibility/uiaccessibilitytraits/1620208-notenabled). This makes VoiceOver read the button as `”dimmed”`, which sufficiently informs the user that the `UIControl` is present but cannot be used yet.
+
+### <a name="toggle-buttons"></a>Use Static Labels for Toggle Buttons
+
+In general, avoid changing the `accessibilityLabel` of elements after it's already been set. If an element's `accessibilityLabel` changes over time, users may think that it is **a different element**. This can be confusing, especially if the element is at the same position on the screen.
+
+If the behavior or meaning of an element, such as a toggle button, changes over time, consider updating the [`accessibilityValue`](https://developer.apple.com/documentation/objectivec/nsobject/1615117-accessibilityvalue) instead. The `accessibilityValue` is read by VoiceOver right after the `accessibilityLabel`. This keeps the user informed that they have the same element focused but it has a different state, and will have a different behavior when activated. 
```

Comment:
This sentence is a bit wordy. It would be good if it could be reduced a bit.

Reply:
I shall try. Thank you for pointing it out. 😬